src/Expr.cc

```diff
@@ -2069,7 +2053,8 @@ bool AssignExpr::TypeCheck(attr_list* attrs)
 		// the script level.
 		return true;
 
-	if ( ((bt1 == TYPE_ENUM) ^ (bt2 == TYPE_ENUM)) )
+	// This should be one of them, but not both (i.e. XOR)
+	if ( (bt1 == TYPE_ENUM) == ! (bt2 == TYPE_ENUM) )
```

Comment:
So - I might be old fashioned, but I actually think that the original statement using the XOR operator is much clearer than this syntax.

I currently plan on undoing this one during the merge.